### DISCUSSION TITLE
Configure on first boot

### DISCUSSION THREAD

**Discussion by AlessandroAmos** [lunarphp/lunar#1013] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
there are some data that are set manually during installation

**Describe the solution you'd like**
create an installation page where the user can define currency, tax and channel data

**Additional context**
an example is what wordpress does.. where the user defines the database, login, site name, etc.

**Comment by glennjacobs** [comment]
So you're suggesting installation screens. It's something we've discussed and there are certainly security concerns with such an approach. 

Naturally, we wouldn't be asking for database credentials, that's not how Laravel works. But I think a wizard-like set of screens hand-holding you through the set-up could be an option.

Perhaps this should be something the hub does when you first login? A bit like a SaaS product would take you through the initial set-up steps.

Then we also need to consider whether this needs to be automated for the command line.

**Comment by glennjacobs** [comment]
Another thought is, if it was a set-up thing in the hub, which was prompted on first login, it could potentially be re-accessed later if the admin wanted to set-up something again.

**Comment by hectorfarahani** [comment]
I am just trying to simplify this feature, and here is my idea:
I assume the goal is to let users know the system needs some data to work properly and provide an easy way for them to insert those data.

As it is mentioned that there are security concerns, and it doesn't match with Laravel's project setup approaches, a widget in the Hub could work better. 

I am not 100% familiar with Lunar but the following points, in general, should be considered.

* The widget could be dismissable (not sure if lunar currently has such a feature for widgets).
* The widget should be visible based on user access/role


An example of this: The WooCommerce admin widget after you install and configure WooCommerce. So, the plugin is already installed and configured, but it says user that you are missing some essential data to make your store work.

![image](https://user-images.githubusercontent.com/6560172/216816486-25b51460-473f-4905-b167-3fd27b32a2a6.png)

![image](https://user-images.githubusercontent.com/6560172/216816574-c1ab814c-d94a-4ac5-8599-e38f83de0189.png)

In this way, we avoid dealing with security concerns, and Laravel's approach, and, there is no need to make it accessible for later configuration.
